pkg/deployments/manager.go

```diff
@@ -174,6 +182,25 @@ func (r *Manager) ResolveDeployment(model string) (string, bool) {
 	return deploy, ok
 }
 
+type k8sClient interface {
+	List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error
+}
+
+// Bootstrap initializes the Manager by retrieving a list of deployments from the k8s cluster and adding them to the Manager's internal state.
+// The k8s client passed should either be non caching or have the cache synced first.
+func (r *Manager) Bootstrap(ctx context.Context, k8sClient k8sClient) error {
```

Comment:
`Manager` has a `Client` field. Can we use that instead of accepting a client arg?